php/errors.php

```diff
@@ -45,19 +45,79 @@
 
 /*  ------------------------------------------------------------------------ */
 
-if (!function_exists('ccxt\error_factory')) {
-    function error_factory($array, $parent) {
-        foreach ($array as $error => $subclasses) {
-            if (!class_exists('ccxt\\'.$error, false)) {
```

Comment:
These checks were intended to fix the issues with the PHP runtime error generation:

- https://github.com/ccxt/ccxt/issues/6007
- https://github.com/ccxt/ccxt/issues/5988
- https://github.com/ccxt/ccxt/issues/5986

So, in order to merge this, we need to make sure that
a) it works when `required` by `composer` in users' PHP projects
b) it works with `include ('ccxt.php');`

I think that maybe runtime generation is not the best idea for PHP after all. I'd rather rewrite it to live in transpile-time, since it would be a win-win:

1. we will resolve the issues with PHP, if any
2. we will save some cycles (including the lib takes too long in runtime)

Ideally, we'd want to have some tests. Lmk your thoughts on this.

Reply:
> I think that maybe runtime generation is not the best idea for PHP after all. I'd rather rewrite it to live in transpile-time in all languages, since it would be a win-win:

I don't think this is the best solution because then we need to have code generation in transpile time and multiple error files. But this is a separate issue to the arguments that the errors take. Also the setup time is really negligible.

> it works when required by composer in users' PHP projects
> it works with include ('ccxt.php'); as well

yeah we should have tests for both these cases, I will add them to this PR (maybe so it is triggered automatically).

what do you think of https://github.com/ccxt/ccxt/pull/6040#issuecomment-546208272, does it make sense?

Reply:
> what do you think of #6040 (comment), does it make sense?

Well, within the logic of this PR, it does make sense, however, we'd rather also control which particular exception type it is... But, tbh, I don't think that controlling global base verbosity using a dict or a static property is a robust design in general. I'd rather keep the `verbosity` as a property of the exchange class or maybe even a class-specific property of each particular error class. Imagine switching verbosity on and off 2000 times per second in an async environment – it will break down. When we replace straightforward inheritance with code that overrides prototype properties in JS – if we don't pay extra care to it, then it usually fails to work on all systems where you would expect it to work, like, say, in browsers, or in frameworks like react or angular or vue or whatever.